### PR TITLE
Add llvm-openmp/9.0.1 recipe

### PR DESCRIPTION
**Pull request by vvilpas**
Specify library name and version:  **llvm-openmp/9.0.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/llvm-openmp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    def configure(self):
+        if not self._supports_compiler():
+            raise ConanInvalidConfiguration("llvm-openmp doesn't support compiler: {} on OS: {}.".
+                                            format(self.settings.compiler, self.settings.os))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}.src".format(self.short_name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+        self._patch_sources()
```
Comment (madebr) [inline recipes/llvm-openmp/all/conanfile.py]:
```suggestion
```
Please move the patching to the `build()` step.

File: recipes/llvm-openmp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        self._cmake = CMake(self)
+        self._cmake.definitions["OPENMP_STANDALONE_BUILD"] = True
+        self._cmake.definitions["LIBOMP_ENABLE_SHARED"] = self.options.shared
+        if self.settings.os == "Linux":
+            self._cmake.definitions["OPENMP_ENABLE_LIBOMPTARGET"] = self.options.shared
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
```
Comment (madebr) [inline recipes/llvm-openmp/all/conanfile.py]:
```suggestion
    def build(self):
        self._patch_sources()
        cmake = self._configure_cmake()
```

**Comment by Morwenn** [comment]
@madebr To be honest I don't know whether it would help, but it does look like something that would help.

**Comment by vvilpas** [comment]
@madebr @danimtb Please, could you review again? Thanks!

File: recipes/llvm-openmp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+class LLVMOpenMpConan(ConanFile):
+    name = "llvm-openmp"
+    short_name = "openmp"
+    description = ("The OpenMP (Open Multi-Processing) specification "
+                   "is a standard for a set of compiler directives, "
+                   "library routines, and environment variables that "
+                   "can be used to specify shared memory parallelism "
+                   "in Fortran and C/C++ programs. This is the LLVM "
+                   "implementation.")
+    license = "BSD-2-Clause"
```
Comment (uilianries) [inline recipes/llvm-openmp/all/conanfile.py:15]:
```suggestion
    license = "Apache-2.0"
```

https://github.com/llvm/llvm-project/blob/llvmorg-9.0.1/openmp/LICENSE.txt

Comment (Croydon) [reply recipes/llvm-openmp/all/conanfile.py:15]:
```suggestion
    license = ("Apache-2.0 WITH LLVM-exception")
```
It isn't Apache-2.0 but Apache 2.0 with LLVM exception

> The LLVM Project is under the Apache License v2.0 with LLVM Exceptions:

https://spdx.org/licenses/LLVM-exception.html

https://spdx.org/licenses/exceptions-index.html
>  Exceptions are added to a license using the License Expression operator, "WITH".

https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60
> Exception "WITH" Operator
Sometimes a set of license terms apply except under special circumstances. In this case, use the binary "WITH" operator to construct a new license expression to represent the special exception situation. A valid <license-expression> is where the left operand is a <simple-expression> value and the right operand is a <license-exception-id> that represents the special exception terms.  
For example, when the Bison exception is to be applied to GPL-2.0+, the expression would be: 
(GPL-2.0+ WITH Bison-exception-2.2)

File: recipes/llvm-openmp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LLVMOpenMpConan(ConanFile):
+    name = "llvm-openmp"
```
Comment (Croydon) [inline recipes/llvm-openmp/all/conanfile.py:7]: Just `openmp` might be the more official name?

**Comment by vvilpas** [comment]
Hi @danimtb! Is there anything else missing in order to merge?

**Pull request merged**